src/expand.ts

```diff
@@ -122,22 +126,24 @@ export async function expandString<T>(input: string | T, opts: ExpansionOptions)
         let i = 0;
         do {
             // TODO: consider a full circular reference check?
-            const expansion = await expandStringHelper(result, opts);
+            const expansion = await expandStringHelper(result, opts, errorHandler);
             result = expansion.result;
             didReplacement = expansion.didReplacement;
             circularReference = expansion.circularReference;
             i++;
         } while (i < maxRecursion && opts.recursive && didReplacement && !circularReference);
 
         if (circularReference) {
-            log.error(localize('circular.variable.reference', 'Circular variable reference found: {0}', circularReference));
+            log.error(localize('circular.variable.reference.full', 'Circular variable reference found: {0}', circularReference));
```

Comment:
Why the changes to the key value for the localize calls here? Is there a reason why we're adding `full` to the end of them all?

Reply:
Ah, I see why now, disregard.

Reply:
needed to add a localized string for the errorHandler with almost the same wording but a bit different so i figured id do that to differentiate